types/cytoscape-fcose/index.d.ts

```diff
@@ -0,0 +1,126 @@
+export = cytoscapeFcose;
+export as namespace cytoscapeFcose;
+
+declare namespace cytoscapeFcose {
+    interface FcoseFixedNodeConstraint {
+        nodeId: string;
+        position: cytoscape.Position;
+    }
+
+    interface FcoseAlignmentConstraint {
+        vertical: Array<[string, string]>;
+        horizontal: Array<[string, string]>;
+    }
+
+    interface FcoseRelativeVerticalPlacementConstraint {
+        top: string;
+        bottom: string;
+        gap: number;
+    }
+
+    interface FcoseRelativeHorizontalPlacementConstraint {
+        left: string;
+        right: string;
+        gap: number;
+    }
+
+    type FcoseRelativePlacementConstraint =
+        | FcoseRelativeVerticalPlacementConstraint
+        | FcoseRelativeHorizontalPlacementConstraint;
+
+    interface FcoseLayoutOptions extends cytoscape.BaseLayoutOptions {
+        name: 'fcose';
+
+        // 'draft', 'default' or 'proof'
+        // - "draft" only applies spectral layout
+        // - "default" improves the quality with incremental layout (fast cooling rate)
+        // - "proof" improves the quality with incremental layout (slow cooling rate)
+        quality?: "default" | "draft" | "proof";
+        // Use random node positions at beginning of layout
+        // if this is set to false, then quality option must be "proof"
+        randomize?: boolean;
+        // Whether or not to animate the layout
+        animate?: boolean;
+        // Duration of animation in ms, if enabled
+        animationDuration?: number;
+        // Easing of animation, if enabled
+        animationEasing?: undefined | "ease-out";
+        // Fit the viewport to the repositioned nodes
+        fit?: boolean;
+        // Padding around layout
+        padding?: number;
+        // Whether to include labels in node dimensions. Valid in "proof" quality
+        nodeDimensionsIncludeLabels?: boolean;
+        // Whether or not simple nodes (non-compound nodes) are of uniform dimensions
+        uniformNodeDimensions?: boolean;
+        // Whether to pack disconnected components - cytoscape-layout-utilities extension should be registered and initialized
+        packComponents?: boolean;
+        // Layout step - all, transformed, enforced, cose - for debug purpose only
+        step?: "all" | "transformed" | "enforced" | "cose";
+
+        /* spectral layout options */
+
+        // False for random, true for greedy sampling
+        samplingType?: boolean;
+        // Sample size to construct distance matrix
+        sampleSize?: number;
+        // Separation amount between nodes
+        nodeSeparation?: number;
+        // Power iteration tolerance
+        piTol?: number;
+
+        /* incremental layout options */
+
+        // Node repulsion (non overlapping) multiplier
+        nodeRepulsion?: number;
+        // Ideal edge (non nested) length
+        idealEdgeLength(edge: any): number;
+        // Divisor to compute edge forces
+        edgeElasticity(edge: any): number;
```

Comment:
```suggestion
        idealEdgeLength?(edge: any): number;
        // Divisor to compute edge forces
        edgeElasticity?(edge: any): number;
```
both of those seem to be optional (from readme defaults):
you can rewrite them to `idealEdgeLength?: (edge: any) => number;`

Reply:
Actually had these changes in my own project. Frogot to copy them. 
Will be ready in a sec.